cmd/cody-gateway/internal/actor/dotcomuser/dotcomuser.go

```diff
@@ -91,6 +98,27 @@ func (s *Source) fetchAndCache(ctx context.Context, token string) (*actor.Actor,
 		s.log.Error("failed to marshal actor",
 			log.Error(err))
 	} else {
+		for _, feature := range codygateway.AllFeatures {
+			rl, ok := act.RateLimits[feature]
+			if !ok {
+				continue
+			}
+
+			// reset the usage cache if:
+			// 1. old actor is not present in the cache
+			// 2. new rl interval is different from old rl interval
+			if oldAct.IsEmpty() || rl.Interval != oldAct.RateLimits[feature].Interval {
+				// get the base limiter for the feature which implements the core rate limiting based on the config
```

Comment:
Can we be more specific than "core rate limiting based on the config"? I mean, it's clear what that is referring too.. I just cannot say off the top of my head what it _means_. e.g. what types of things are we limiting based on the config, and what we would call these "other" types of rate limiters (dependent on user subscriptions, etc.)

Would it be helpful to try and introduce new terms here? Like "anti-abuse limiting" or "subscription-usage limiting" or something?

```suggestion
				// Get the base limiter for the feature which implements the core rate limiting based on the config.
```